redkubes/otomi-core
# feat: drone activation and otomi version upgrade with cypress

### Implements:
https://app.zenhub.com/workspaces/dev-5e7e84dddc966f05a627a19b/issues/gh/redkubes/otomi-e2e/30

### Description:
This PR adds features for drone activation and otomi version upgrade with cypress. User doesn't need to activate drone manually. Otomi drone activation job does this work for the user.

Is paired with: https://github.com/redkubes/otomi-console/pull/311

### Changes Made:
Add drone activation job to activate repository
Add e2e job to upgrade otomi version

### Testing:
- Open [Deploy Otomi](https://github.com/redkubes/otomi-core/actions/workflows/integration.yml) page 
- Run workflow with `feat/e2e` branch
- After the cluster created check drone if it's activated

- Run workflow with `feat/e2e` branch and `upgrade` Otomi installation profile
- After the cluster created check drone if it's activated
- After the initial pipelines finished check otomi version if it's setted as main in otomi settings page

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.
